spark/Dockerfile

```diff
@@ -4,22 +4,38 @@ LABEL maintainer="Stackable GmbH"
 # https://github.com/hadolint/hadolint/wiki/DL4006
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
+
 RUN microdnf update && \
     microdnf install java-11-openjdk-headless --nodocs && \
     microdnf install tar gzip zip && \
     microdnf install shadow-utils && \
     # required for spark startup scripts
     microdnf install procps && \
     microdnf install hostname && \
+    microdnf install python3 python3-pip && \
     microdnf clean all
 
+RUN ln -s /usr/bin/python3 /usr/bin/python
+RUN ln -s /usr/bin/pip3 /usr/bin/pip
+
 ENV JAVA_HOME=/usr/lib/jvm/jre-11
+ENV SPARK_HOME=/stackable/spark
+ENV PATH=$SPARK_HOME:$PATH:/bin:$JAVA_HOME/bin:$JAVA_HOME/jre/bin
+ENV PYSPARK_PYTHON=/usr/bin/python
+ENV PYTHONPATH=$SPARK_HOME/python:$PYTHONPATH
 
 COPY spark/stackable /stackable
+COPY spark/requirements.txt /tmp/requirements.txt
+
 RUN groupadd -r stackable --gid=1000 && \
     useradd -r -g stackable --uid=1000 stackable && \
     chown -R stackable:stackable /stackable
 
+# pyspark dependencies
+RUN pip install --upgrade pip
+RUN pip install --no-cache-dir \
```

Comment:
Since spark runs as the `stackable` user, you should be able to use `pip install --user ...` to install under the user's home instead of system wide. But I guess we also need to set the HOME path which is not being currently done.